File: Lecture003 If-Else, While loop & Lots of Patterns/javaFiles/Pattern22.java
package com.company;

import java.util.Scanner;

public class Pattern22 {
    // Pattern 22
    public static void main(String[] args) {
        /*
        Pattern 22
                       1
                       0 1
                       1 0 1
                       0 1 0 1
                       1 0 1 0 1
         */
        // Creating Scanner Object For taking inputs from the USER
        Scanner sc = new Scanner(System.in);
        // Take Number of ROWS and COLUMNS from the USER
        System.out.print(" Enter Number of ROWS for the Pattern: ");
        int n = sc.nextInt();
        pattern(n);
    }

    static void pattern(int n) {
        // Procedure to print the Required Pattern
        for (int row = 1; row <= n; row++) {
            for (int column = 1; column <= row; column++) {
                if((row + column) % 2 == 0)
                    System.out.print(0 + " ");
                else
                    System.out.print(1 + " ");
            }
            System.out.println();
        }
    }
}

File: 003_If-Else_While_loop_and_Lots_of_Patterns/Swift/patterns/Pattern22.swift

import Foundation

// Function to print the specified pattern
func printPattern(_ n: Int) {
    for row in 1...n {
        for column in 1...row {
            let value = (row + column) % 2 == 0 ? 0 : 1
            print("\(value) ", terminator: "")
        }
        print("")
    }
}

// Taking user input for the number of rows
print("Enter Number of ROWS for the Pattern: ", terminator: "")
if let input = readLine(), let n = Int(input) {
    printPattern(n)
} else {
    print("Invalid input")
}
